Bibliography labels match links

jackwyand opened the pull request:
Contributes to #4046 

Not 100% sure this is what was meant when the bibliography labels should be text, let me know if anything should be changed.

Screenshot of output: 
![image](https://github.com/user-attachments/assets/cb35eefd-c452-4107-a672-959cc40663a3)

smiths commented:
@jackwyand I like the appearance of your proposed Reference List.  A tweak would be to put the reference name between square brackets.  Getting more fancy would have the reference itself indented, like in this [Overleaf example](https://www.overleaf.com/learn/latex/Biblatex_citation_styles).

@jackwyand the checks failed because your change modifies the examples.  You also need to update the [stable version of the output files](https://github.com/JacquesCarette/Drasil/wiki/Workflow#updating-stable-folder).

jackwyand commented:
I changed the formatting and updated `stable`, I definitely like this better :)
![image](https://github.com/user-attachments/assets/99d15d4d-dbd4-41d6-b261-6e9f5585f323)

I hope there aren't any issues with changing `stable`, the [workflow](https://github.com/JacquesCarette/Drasil/wiki/Workflow#updating-stable-folder-files) mentions 
>Commit and push these changes before a PR. This will help avoid Continuous Integration (CI) build failures or errors.

Considering this PR already existed, will that be an issue?

smiths reviewed (approved):
The changes to stable look good to me.  The new generated reference lists look much better than the old lists and they use labels, not numbers.  @JacquesCarette or @balacij will have to review how the generation was done, but the output looks good.

balacij commented on code/stable/dblpend/SRS/HTML/DblPend_SRS.html:
```diff
@@ -3940,53 +3940,44 @@ <h1>Values of Auxiliary Constants</h1>
+        <dl class="reference-list">
+          <dt class="reference-label">[<b>hibbeler2004</b>]</dt>
+          <dd>
+            Hibbeler, R. C. <em>Engineering Mechanics: Dynamics</em>. Pearson Prentice Hall, 2004. Print.
```
Just a minor issue: we need to add the `id` attributes to the `dt` notes so that hyperlinks work. Everything else looks perfect :)
